src/util/hashContext.ts

```diff
@@ -1,16 +1,31 @@
 import * as murmurHash3 from 'murmurhash3js'
 import globals from '../globals'
 import { ID } from '../constants'
+import _ from 'lodash'
 
 // util
 import { escapeSelector } from './escapeSelector'
 import { pathToContext } from './pathToContext'
 import { Context, Path } from '../types'
+import { stripTags } from './stripTags'
+import { stripEmojiWithText } from './stripEmojiWithText'
+import { singularize } from './singularize'
+import { trim } from './trim'
 
 const SEPARATOR_TOKEN = '__SEP__'
 
+/** Converts a string to lowecase. */
+const lower = (s: string) => s.toLowerCase()
+
 /** Encode the thoughts (and optionally rank) as a string. */
 export const hashContext = (thoughts: Context | Path, rank?: number): string => (globals.disableThoughtHashing ? ID : murmurHash3.x64.hash128)(pathToContext(thoughts)
   .map(thought => thought ? escapeSelector(thought) : '')
+  .map(thought => _.flow([
+    stripTags,
+    lower,
+    trim,
+    stripEmojiWithText,
+    singularize,
+  ])(thought))
```

Comment:
The composition of string modifiers that are fed into  murmurHash should be abstracted into a function that both `hashContext` and `hashThought` use. Otherwise it will be too easy for one of them to get out of sync.

```js
const compress = _.flow([
    stripTags,
    lower,
    trim,
    stripEmojiWithText,
    singularize,
  ])
```

Reply:
Sure.